x/clock/abci.go

```diff
@@ -1,42 +1,111 @@
 package clock
 
 import (
-	"log"
 	"time"
 
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	"github.com/cosmos/cosmos-sdk/telemetry"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	"github.com/CosmosContracts/juno/v19/x/clock/keeper"
 	"github.com/CosmosContracts/juno/v19/x/clock/types"
 )
 
+var endBlockSudoMessage = []byte(types.EndBlockSudoMessage)
+
 // EndBlocker executes on contracts at the end of the block.
 func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
 	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)
 
-	message := []byte(types.EndBlockSudoMessage)
-
+	logger := k.Logger(ctx)
 	p := k.GetParams(ctx)
 
-	errorExecs := make([]string, len(p.ContractAddresses))
+	// Get all contracts
+	contracts, err := k.GetAllContracts(ctx)
+	if err != nil {
+		logger.Error("Failed to get contracts", "error", err)
+		return
+	}
+
+	// Track errors
+	errorExecs := make([]string, len(contracts))
+	errorExists := false
 
-	for idx, addr := range p.ContractAddresses {
-		contract, err := sdk.AccAddressFromBech32(addr)
+	// Function to handle contract execution errors. Returns true if error is present, false otherwise.
+	handleError := func(err error, idx int, contractAddress string) bool {
```

Comment:
is there a way we can not have this be an anon function & make it a priv function?

re-assigning the memory every block scares me for a mem leak

Reply:
On it! The anon function was used to reduce the number of parameters we need to pass to the handleError function.